### DISCUSSION TITLE
Using PML for text modality

### DISCUSSION THREAD

**Discussion by vemchance** [KevinMusgrave/pytorch-metric-learning#707] [Q&A] [answered]
Hi there, thanks again for the brilliant library!

I'm more of a computer vision person myself but I'm hoping to use parts of this library for a text and image problem. I've not been able to successfully adapt any of the examples to text-only (e.g., using SentenceBERT or even just BERT embeddings for testing purposes) and wondered if anyone else had used the library for text?

I think my primary problem is I'm not familiar enough with NLP models to understand how to tokenized data should be loaded for the testing phase for this library in some of the examples (e.g., with [this](https://github.com/KevinMusgrave/pytorch-metric-learning/blob/master/examples/notebooks/TripletMarginLossMNIST.ipynb) example I can get through the training phase but on testing I don't think the input to the model quite work there). 

I am probably going to go back to a much simpler implementation than the examples and try from there, but I wondered if anyone had used the library for something beside images and had any tips, tricks or general advice to point me in the right direction?

**Comment by KevinMusgrave** [comment, marked as answer]
Some potentially relevant examples that I think use this library, but maybe not for the testing phase:

- https://github.com/JohnGiorgi/DeCLUTR
- https://github.com/cambridgeltl/sapbert
- https://github.com/cambridgeltl/mirror-bert
- https://github.com/thunlp/RE-Context-or-Names
- https://github.com/jzhang38/LPD

**Reply by vemchance** [reply]
Thanks for the reply! I thought I'd put a note in for anyone else - based on the suggested libraries, I was able to just modify the BaseTester functions in your library to accept a dictionary for BERT .etc, which ended up being the easiest solution as the problem seemed to be moving data between devices from a dictionary. It works really well - took me awhile to figure out but looking through these helped!
